### PR TITLE
fix public view

### LINKED ISSUE #33803: BUG: public link image share, footer overlaps with scrollbar:
image

### LINKED ISSUE #33858: sharing view needs a bit of rework
multiple things:
- list view/gallry view toggle looks out of place
- the three dots in the top right are mostly unvisible
image

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/server/issues/33858
Close https://github.com/nextcloud/server/issues/33803

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Comment by max-nextcloud** [comment]
thanks for looking into this!
Haven't had time yet to try it out.

File: apps/files_sharing/css/public.scss [hunk 1]
```diff
@@ -27,8 +30,13 @@
 }
 
 #imgframe img {
-	max-height: 100% !important;
+	max-height: calc(100vh - var(--header-height) - #{$footer-height} - #{$download-button-section-height}) !important;
```
Comment (skjnldsv) [inline apps/files_sharing/css/public.scss:33]: Oh wow :see_no_evil:

**Review by PVince81 — APPROVED** [review]
:+1:

File: apps/files_sharing/css/public.scss [hunk 2]
```diff
@@ -27,8 +30,13 @@
...
 	max-width: 100% !important;
+	width: unset !important;
+}
+
+#imgframe video {
+	max-height: calc(100vh - var(--header-height) - #{$footer-height} - #{$download-button-section-height});
```
Comment (skjnldsv) [inline apps/files_sharing/css/public.scss:39]: Is it still relevant?
Considering we're using the viewer now to render single files shares?

File: apps/files_sharing/css/public.scss (hunk 1 above)
Comment (szaimen) [reply apps/files_sharing/css/public.scss:33]: Do you have better ideas? Other solutions didnt work in my testing...

File: apps/files_sharing/css/public.scss (hunk 2 above)
Comment (szaimen) [reply apps/files_sharing/css/public.scss:39]: Yes, this was needed to make it work with any screen size...

Comment (skjnldsv) [reply apps/files_sharing/css/public.scss:39]: SHould it not be handled widely then? Make the viewer container takign the full size?
Not just `video` or `img` tags

Comment (szaimen) [reply apps/files_sharing/css/public.scss:39]: Btw: public audio playback is currently broken IIRC: https://github.com/nextcloud/server/issues/34027

Comment (max-nextcloud) [reply apps/files_sharing/css/public.scss:39]: I'm working on a separate PR for audio playback here #34051 - would style that after this is merged.

**Review by max-nextcloud — APPROVED** [review]

**Pull request merged**